File: clients/messagedb_ex/test/system_test.exs
defmodule MessagedbEx.SystemTest do
  use ExUnit.Case, async: false
  import MessagedbEx.TestHelper

  setup do
    {client, namespace_id, _token} = create_test_namespace("system")

    on_exit(fn ->
      cleanup_namespace(client, namespace_id)
    end)

    %{client: client}
  end

  test "SYS-001: Get server version", %{client: client} do
    assert {:ok, version, _client} = MessagedbEx.system_version(client)
    assert is_binary(version)
    # Should match semver pattern
    assert String.match?(version, ~r/^\d+\.\d+\.\d+/)
  end

  test "SYS-002: Get server health", %{client: client} do
    assert {:ok, health, _client} = MessagedbEx.system_health(client)
    assert is_map(health)
    assert health.status == "ok"
  end
end

File: clients/eventodb-node/tests/system.test.ts
import { describe, test, expect } from 'vitest';
import { getEventoDBURL } from './helpers.js';
import { MessageDBClient } from '../src/client.js';

const ADMIN_TOKEN = process.env.EVENTODB_ADMIN_TOKEN;

describe('SYS Tests', () => {
  test('SYS-001: Get server version', async () => {
    // System endpoints may require authentication in production mode
    const client = new MessageDBClient(getEventoDBURL(), {
      token: ADMIN_TOKEN
    });
    
    const version = await client.systemVersion();

    expect(typeof version).toBe('string');
    // Should match semver pattern (e.g., "1.3.0")
    expect(version).toMatch(/^\d+\.\d+\.\d+/);
  });

  test('SYS-002: Get server health', async () => {
    // System endpoints may require authentication in production mode
    const client = new MessageDBClient(getEventoDBURL(), {
      token: ADMIN_TOKEN
    });
    
    const health = await client.systemHealth();

    expect(health).toBeDefined();
    expect(health.status).toBeDefined();
    expect(typeof health.status).toBe('string');
  });
});
